sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala

```diff
@@ -459,6 +459,19 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     }
   }
 
+  test("UnresolvedMapObjects should support interpreted execution") {
+    val inputExpr = Literal(1)
+    val func = (expr: Expression) => expr
+    val customCls: Option[Class[_]] = None
+
+    checkError(
+      exception = intercept[SparkUnsupportedOperationException] {
+        UnresolvedMapObjects(func, inputExpr, customCls).dataType
```

Comment:
Could you try to trigger the error from user code. BTW, the error classes should be assigned to user-facing errors.

Reply:
I couldn't do something interesting in that matters. Should I proceed with SparkException.internalError() approach?